src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridItemModel.cs

```diff
@@ -96,23 +102,88 @@ protected TreeDataGridItemModel()
                         // gets disposed when the model is deactivated. This is important to
                         // understand for the model and child activation/deactivation relationships.
                         model._childrenObservableSerialDisposable.Disposable = null;
-                        CleanupChildren(model._children);
 
                         if (isInitializing)
                         {
                             model._childrenObservableSerialDisposable.Disposable = model.ChildrenObservable
                                 .OnUI()
-                                .DisposeMany()
-                                .SubscribeWithErrorLogging(changeSet => model._children.ApplyChanges(changeSet));
+                                .SubscribeWithErrorLogging(changeSet =>
+                                {
+                                    UpdateChildren(model._children, changeSet);
+                                });
                         }
                     }, onCompleted: static (_, model) => CleanupChildren(model._children));
             }
         });
     }
 
-    private static void CleanupChildren(ObservableList<TModel> children)
+    private static void UpdateChildren(SourceCache<TModel, TKey> cache, IChangeSet<TModel, TKey> changeSet)
```

Comment:
This isn't working in case one of the children is removed while the parent is collapsed.
We still have the child, but when we start processing updates again we only receive Adds for every element in the current collection, no Removes for our existing items since we had stopped listening.

So we would need to have special handling for when we make the subscription that checks the current SourceCache against the current contents and updates, then goes to listen changes from there.